src/dmd/dsymbolsem.d

```diff
@@ -5989,3 +5986,95 @@ void aliasSemantic(AliasDeclaration ds, Scope* sc)
             ScopeDsymbol.multiplyDefined(Loc.initial, sx, ds);
     }
 }
+
+/**
+ * build a shim function around the compound dtor that accepts an argument
+ *  that is used to implement the deleting C++ destructor
+ *
+ * Params:
+ *  cldev = the class that contains the destructor to wrap
+ *  sc = the scope in which to analyze the new function
+ *
+ * Returns:
+ *  the shim destructor, semantically analyzed and added to the class as a member
+ */
+private FuncDeclaration buildWindowsCppDtor(ClassDeclaration cldec, Scope* sc)
+{
+    auto dtor = cldec.dtor;
+    if (!dtor)
+        return null;
+
+    Parameter param = new Parameter(STC.undefined_, Type.tuns32, Identifier.idPool("del"), null);
+    Parameters* params = new Parameters;
+    (*params).push(param);
+    auto ftype = new TypeFunction(params, Type.tvoidptr, false, LINK.cpp, dtor.storage_class);
+    auto func = new DtorDeclaration(dtor.loc, dtor.loc, dtor.storage_class, Id.cppdtor);
+    func.type = ftype;
+    if (dtor.fbody)
+    {
+        Loc loc = dtor.loc;
+        auto stmts = new Statements;
+        auto call = new CallExp(loc, dtor, null);
```

Comment:
Why don't these calls receive 'this' as an argument?

Reply:
`dtor` is a member function, so semantic figures this out. Not sure I need the ubiquitous hack casting away const/immutable, though.

Reply:
I saw that everywhere; just rolled with it ;)